### PR TITLE
Init numeric filters

### PR DESCRIPTION
**Pull request by alexvpickering**
# Background
#### Link to issue 

https://biomage.atlassian.net/jira/software/c/projects/BIOMAGE/boards/1?modal=detail&selectedIssue=BIOMAGE-1253

#### Link to staging deployment URL 

https://ui-olivergibson-worker218.scp-staging.biomage.net/

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [x] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Review by gerbeldo — COMMENTED** [review]
Looks good to me! 

Some random comments/question:
1. weren't we using snake case for functions?
2. double assignment in a single line should be illegal (mock_gene_results...).
3. animal photo is missing 👀

File: r/R/worker_helpers.R [hunk 1]
```diff
@@ -143,11 +143,44 @@ getClusters <- function(type, resolution, data) {
...
-  if (!is.null(filter)) {
-    gene_results <- gene_results[grepl(filter, gene_results$gene_name, ignore.case = TRUE), ]
+applyFilters <- function(gene_results, filters) {
+
+  filter_columns <- sapply(filters, `[[`, 'columnName')
+
+  # apply gene text filter
+  gene.idx <- which(filter_columns == 'gene_names')[1]
+  if (!is.na(gene.idx)) {
+    gene <- filters[[gene.idx]]$expression
```
Comment (ogibson) [inline r/R/worker_helpers.R:153]: It took me a second to figure out that expression referred to regex. I was like "why would you have expression values here". Maybe we change it? TBH it's pretty obvious its regex, but just to clarify?

Whatever you think is fine.

File: r/R/worker_helpers.R [hunk 2]
```diff
@@ -143,11 +143,44 @@ getClusters <- function(type, resolution, data) {
...
+    if (!column %in% numeric_columns) next()
+    filter <- filters[[idx]]
+
+    gene_results <- applyNumericFilter(gene_results, column, filter$comparison, filter$value)
+  }
+
+  return(gene_results)
+}
+
+applyNumericFilter <- function(gene_results, column, comparison, value) {
```
Comment (ogibson) [inline r/R/worker_helpers.R:171]: no equal?

Comment (ogibson-biomage) [reply r/R/worker_helpers.R:171]: >= / <=

Comment (ogibson-biomage) [reply r/R/worker_helpers.R:171]: >= / <=

Comment (ogibson) [reply r/R/worker_helpers.R:171]: Wont this lead to some border case? I know exact value will be very hard to hit but still. Something like >= / <=

**Review by ogibson — APPROVED** [review]

**Comment by ogibson** [comment]
Tested the list of genes search functionality and it's ok. Same for DE. Can't test numeric filtering?

File: r/R/worker_helpers.R (hunk 1 above)
Comment (alexvpickering) [reply r/R/worker_helpers.R:153]: `expression` is what the UI is sending as a request. Agree I'm not a fan

File: r/R/worker_helpers.R (hunk 2 above)
Comment (alexvpickering) [reply r/R/worker_helpers.R:171]: I think the motivation was to simplify things and also e.g. logFC > 0 implies upregulated not logFC >= 0 and then for everything else, as you said, hitting exact values won't happen so it's not necessary.

We can have four cases if you think it's a good idea/justified though.

**Comment by alexvpickering** [comment]
> Looks good to me!
> 
> Some random comments/question:
> 
> 1. weren't we using snake case for functions?

at least for the worker we agreed to go camelCase for function names and snake_case for everything else going forward.

> 2. double assignment in a single line should be illegal (mock_gene_results...).

I'll change if your not a fan but doesn't bother me :)

> 3. animal photo is missing eyes

┈┈╭━╱▔▔▔▔╲━╮┈┈┈
┈┈╰╱╭▅╮╭▅╮╲╯┈┈┈
╳┈┈▏╰┈▅▅┈╯▕┈┈┈┈
┈┈┈╲┈╰━━╯┈╱┈┈╳┈
┈┈┈╱╱▔╲╱▔╲╲┈┈┈┈
┈╭━╮▔▏┊┊▕▔╭━╮┈╳
┈┃┊┣▔╲┊┊╱▔┫┊┃┈┈
┈╰━━━━╲╱━━━━╯┈╳

File: r/tests/testthat/test-worker_helpers.R [hunk 3]
```diff
@@ -0,0 +1,68 @@
+mock_gene_results <- function() {
+    data(exprs, package = 'presto')
+    data(y, package = 'presto')
+
+    result <- presto::wilcoxauc(exprs, y)
+    result <- result[result$group == "A", ]
+    rownames(result) <- result$gene_names <- result$feature
```
Comment (alexvpickering) [inline r/tests/testthat/test-worker_helpers.R]:
```suggestion
    rownames(result) <- result$feature
    result$gene_names <- result$feature
```

**Comment by alexvpickering** [comment]
> Tested the list of genes search functionality and it's ok. Same for DE. Can't test numeric filtering?

It's not implemented in the UI yet. I suppose you could mock the request or check out the tests which do cover it.

**Pull request merged**